src/unit.rs

```diff
@@ -163,12 +160,69 @@ impl Unit {
     }
 }
 
-/// Perform a connection. Used recursively for redirects.
+/// Perform a connection. Follows redirects.
 pub(crate) fn connect(
-    unit: Unit,
+    mut unit: Unit,
+    use_pooled: bool,
+    mut body: SizedReader,
+    mut previous: Vec<String>,
```

Comment:
This is now state entirely internal to the `connect` fn, so it can be moved to a local variable.

Reply:
It's not entirely internal state - `send_prelude` needs to know if this is a redirect or not. But it can do that just as easily with a slice instead.

Reply:
Oh wait, I just realized you meant this should only ever be called with `vec![]`. Yes, it can be removed from `connect`, just not `connect_inner`.